### PR TITLE
Cleaned up std.algorithm.joiner's examples

### PR DESCRIPTION
**Pull request by JackStouffer**
The current examples are pretty hard to follow. What I think happened is someone created some tests that were never meant to be examples, and then sometime later someone added the ddoc comment on it.

My version uses UFCS and reorganizes everything to be much friendlier.

### CODE REVIEW

**Comment by CyberShadow** [comment]
LGTM

**Comment by Hackerpilot** [comment]
Auto-merge toggled on
